test/instrumentation/modules/@aws-sdk/fixtures/use-s3-v3.js

```diff
@@ -0,0 +1,238 @@
+  command = new DeleteObjectCommand({ Bucket: bucketName, Key: key })
+  data = await s3Client.send(command)
+  log.info({ data }, 'deleteObject')
+
+  if (!bucketIsPreexisting) {
+    // https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/classes/deletebucketcorscommand.html
+    command = new DeleteBucketCommand({ Bucket: bucketName })
+    data = await s3Client.send(command)
+    log.info({ data }, 'deleteBucket')
+  }
+}
+
+// Return a timestamp of the form YYYYMMDDHHMMSS, which can be used in an S3
+// bucket name:
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+function getTimestamp () {
+  return (new Date()).toISOString().split('.')[0].replace(/[^0-9]/g, '')
+}
+
+// ---- mainline
+
+function main () {
+  // Config vars.
+  const region = process.env.TEST_REGION || 'us-east-2'
+  const endpoint = process.env.TEST_ENDPOINT || null
+  const bucketName = process.env.TEST_BUCKET_NAME || TEST_BUCKET_NAME_PREFIX + getTimestamp()
+
+  // Guard against any bucket name being used because we will be creating and
+  // deleting objects in it, and potentially *deleting* the bucket.
+  if (!bucketName.startsWith(TEST_BUCKET_NAME_PREFIX)) {
+    throw new Error(`cannot use bucket name "${bucketName}", it must start with ${TEST_BUCKET_NAME_PREFIX}`)
+  }
+
+  const s3Client = new S3Client({
+    region,
+    endpoint,
+    // In Jenkins CI the endpoint is "http://localstack:4566", which points to
+    // a "localstack" docker container on the same network as the container
+    // running tests. The aws-sdk S3 client defaults to "bucket style" URLs,
+    // i.e. "http://$bucketName.localstack:4566/$key". This breaks with:
+    //    UnknownEndpoint: Inaccessible host: `mahbukkit.localstack'. This service may not be available in the `us-east-2' region.
+    //        at Request.ENOTFOUND_ERROR (/app/node_modules/aws-sdk/lib/event_listeners.js:530:46)
+    //        ...
+    //    originalError: Error: getaddrinfo ENOTFOUND mahbukkit.localstack
+    //        at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:66:26) {
+    //      errno: 'ENOTFOUND',
+    //      code: 'NetworkingError',
+    //      syscall: 'getaddrinfo',
+    //      hostname: 'mahbukkit.localstack',
+    //
+    // It *works* with common localstack usage where the endpoint uses
+    // *localhost*, because "$subdomain.localhost" DNS resolution still resolves
+    // to 127.0.0.1.
+    //
+    // The work around is to force the client to use "path-style" URLs, e.g.:
+    //    http://localstack:4566/$bucketName/$key
+    forcePathStyle: true
+  })
+
+  // Ensure an APM transaction so spans can happen.
+  const tx = apm.startTransaction('manual')
+
+  useS3v3(s3Client, bucketName).then(
+    function () {
+      tx.end()
+      process.exitCode = 0
+    },
+    function () {
+      tx.setOutcome('failure')
+      tx.end()
+      process.exitCode = 1
+    }
+  )
```

Comment:
This script was hanging at the end ... which was a timeout on the HTTP client that `s3Client` is using (at least I strongly suspect so). I fixed that with:

```diff
@@ -225,11 +225,13 @@ function main () {
   useS3v3(s3Client, bucketName).then(
     function () {
       tx.end()
+      s3Client.destroy()
       process.exitCode = 0
     },
     function () {
       tx.setOutcome('failure')
       tx.end()
+      s3Client.destroy()
       process.exitCode = 1
     }
   )
```

Reply:
I've added it. Thanks